pkg/controller/messaginginfra/broker/controller.go

```diff
@@ -102,15 +109,27 @@ func (b *BrokerController) ReconcileBrokers(ctx context.Context, logger logr.Log
 				if err != nil {
 					return err
 				}
+				delete(hosts, toHost(&brokers.Items[i]))
 				toDelete--
 			}
 		} else {
 			logger.Info("No changes to broker pool")
 		}
+
+		// Update discoverable brokers
+		newHosts := make([]string, 0)
```

Comment:
you've called the variable newHosts, but it also contains the existing hosts.  Should it be called allHosts?